src/Compute/Compute/Extension/GetAzureVMExtensionCommand.cs

```diff
@@ -65,16 +65,28 @@ public override void ExecuteCmdlet()
 
             ExecuteClientAction(() =>
             {
-                if (Status.IsPresent)
+                if (!string.IsNullOrEmpty(Name))
                 {
-                    var result = this.VirtualMachineExtensionClient.GetWithInstanceView(this.ResourceGroupName, this.VMName, this.Name);
-                    WriteObject(result.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName));
+                    if (Status.IsPresent)
+                    {
+                        var result = this.VirtualMachineExtensionClient.GetWithInstanceView(this.ResourceGroupName, this.VMName, this.Name);
+                        WriteObject(result.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName));
+                    }
+                    else
+                    {
+                        var result = this.VirtualMachineExtensionClient.GetWithHttpMessagesAsync(this.ResourceGroupName,
+                            this.VMName, this.Name).GetAwaiter().GetResult();
+                        WriteObject(result.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName));
+                    }
                 }
                 else
                 {
-                    var result = this.VirtualMachineExtensionClient.GetWithHttpMessagesAsync(this.ResourceGroupName,
-                        this.VMName, this.Name).GetAwaiter().GetResult();
-                    WriteObject(result.ToPSVirtualMachineExtension(this.ResourceGroupName, this.VMName));
+                    var result = this.VirtualMachineExtensionClient.GetWithInstanceView(this.ResourceGroupName, this.VMName);
```

Comment:
So for list case, it always lists instance view, regardless of -Status parameter?

Reply:
Yes, for now yes, honestly, I couldn't fine where I should override or change GetWithHttpMessagesAsync....

Reply:
@maddieclayton, It should be fixed here to: https://github.com/lwajswaj/azure-powershell/commit/1fc53716994dc3f5ff922ddb5eb749670b5004ed

Reply:
@hyonholee This should be fixed now.

@lwajswaj Thanks! Looks good. Works as expected and tests are passing.

Reply:
Nice!